### ISSUE TITLE
Publication plan

### ISSUE THREAD

**Issue by mesarcik** [mesarcik/ROAD#1]
# Background
This document serves as a plan for the final publication in my PhD. This work is a continuation of the novelty-based detection scheme proposed in the last two publication. Here we discuss some of the foreseeable difficulties in both constructing the dataset as well as training models for anomaly detection.  The following section are in order of completion, i.e. creating the dataset needs to happen before training the models. 
    
# Dataset creation
We need to translate the [spreadsheet-based](https://docs.google.com/spreadsheets/d/1XYrxrh_iyW5SRRHVqCrA5i16e8gKCzYA/edit#gid=996303470) labels to spectrograms that can be used for training the anomaly detection based models. I began this process late last year, but discovered a bug in the station names extraction code.  The following todo lists serves as a guide of the steps needed to 

## TODO

- [x] Pull Jorrit's patch
- [x] Separate the anomalies from autocorrelations and cross correlations 
    - Maybe it sufficient to detect anomalies in the autocorrelation spectrum because then we can assume all baselines that contain that station will also contain the anomaly.
- [x] Ensure that there is only 1 type of anomaly per class. 
- [x] There should be only 1 training set for all the anomalies, this training set should contain no anomalies 
    -  Should the training set only consist of autocorrelations? This means we have no phase information? 
- [x] To detect only one specific anomaly we would need to sample anomalies from the other dataset. The scheme I propose is shown below. 
    - One of the problems with this approach is that if not all other known anomalies are contained within the training set, then we may detect the unseen anomaly as the class we are trying to detect 
    - For example, if we are training an oscillating tile detector and used only the two classes shown in the diagram below, then if the model were to be exposed to lightening then it may say it is an oscilating tile?

## Other considerations
- Focus on 4 anomalies initially 
    - High noise element
    - Data loss
    - Oscillating tile
    - Scintilation
    -  then expand to the more rare classes that may require higher resolution data
    
- Unlike the RFI work, the labels only need to be on a per-spectrogram level
- There will be multiple testing classes but 1 training class (that contains only "normal" LOFAR data).

## Structure

``` bash
LOFARAD/

|-- train
|   |-- crosscorrelations
|   |  |   |-- Baseline ID
|   |  |   |-- Baseline ID
|   |  |   |-- ...
|   |-- autocorrelations
|   |  |   |-- Baseline ID
|   |  |   |-- Baseline ID
|   |  |   |-- ...

|-- test
|   |-- high_noise
|   |   |-- crosscorrelations
|   |   |    |-- Baseline ID
|   |   |    |-- Baseline ID
|   |   |    |-- ...
|   |   |-- autocorrelations
|   |   |    |-- Baseline ID
|   |   |    |-- Baseline ID
|   |   |    |-- ...
|   |-- oscillating_tile
|   |   |-- crosscorrelations
|   |   |    |-- Baseline ID
|   |   |    |-- Baseline ID
|   |   |    |-- ...
|   |   |-- autocorrelations
|   |   |  |   |-- Baseline ID
|   |   |  |   |-- Baseline ID
|   |   |  |   |-- ...
|-- <etc etc>

```

# Training procedure      
- Multi class anomaly detection methodology for LOFAR 
    - Train excluding a single class and all others, evaluate how well it detects that class 
    - This means to detect 4 different anomalies we need 4 different models that are trained on completely different data (similar how we evaluated on MNIST in the NLN paper) 
    - This may also give an opportunity for efficient computing as the models are run in parallel it could offer some interesting implementation details
    

![trianing_diagram](https://user-images.githubusercontent.com/8803403/177991913-5f1bcc87-464e-4daa-b08a-963d9565d8d6.png)

# Model selection
- Evaluate NLN on this dataset 
    - Limitations of the method, we predict a scalar for a given input spectrogram, using AE's for this means that we need to average the Reconstruction error which will probably cause poor sensitivity
    - I think singular mapping using self-supervised or contrastive losses will be best
- Investigate alternative self-supervised losses that produce a scalar per spectrogram without averaging the reconstruction error
    - We have extra information about the baseline (get a model to predict which baseline an input is from)
    - We can include another subtask? 
    - Do we use a contrastive loss? Are the spectrograms sufficiently different to do so? Maybe if we include a contrastive loss per baseline?  
    - Get the model to fill in the blanks like the paper we referenced for our NLN paper? 

# Other thoughts
- Timeline? When do i need to start writing my thesis?
- Potential venues, short conference paper at SPIE, journal paper for MNRAS?
